packages/uows-client-generator/src/construct-functions/constructFunctions.ts

```diff
@@ -42,7 +42,15 @@ export const createFunctTemplate = (
 
   return `    public async ${functName}(${wrapperArgString}) : Promise<any> {
                   const argsObj = this.makeArgsObj('${functName}'${makeArgObjArgs});
+                  const requestId = createHash('sha1').update('${functName}' + JSON.stringify(argsObj)).digest('base64');
+
+                  const activeUowsRequest = this.activeUowsRequests[requestId];
+                  if (activeUowsRequest) {
+                    return activeUowsRequest;
+                  }
+
                   const refinedResult = this.client['${functName}Async'](argsObj).then((result: any) => {
+                      delete this.activeUowsRequests[requestId];
```

Comment:
I think this should also be done in the promises `.catch` to avoid perma-cache-ing failing responses